src/app/components/columns-modify-dialog.component.spec.ts

```diff
@@ -0,0 +1,117 @@
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      providers: [
+        ColumnsModifyDialogComponent,
+        { provide: MatDialogRef, useValue: mockMatDialogRef },
+        { provide: MAT_DIALOG_DATA, useValue: mockMatDialogData }
+      ]
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ColumnsModifyDialogComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should run', () => {
+    expect(component).toBeTruthy();
+  });
+
+  test('optionsColumnValue should add "options." to any provided string', () => {
+    expect(component.optionsColumnValue('nameColumn')).toEqual('options.nameColumn');
+  });
+
+  describe('columnToLabel', () => {
+    it('should get the label of the specified column', () => {
+      expect(component.columnToLabel('name' as ColumnKey<object, object>))
+        .toEqual('Name');
+    });
+
+    it('should return the stringified key in case of non-existing label', () => {
+      expect(component.columnToLabel('nonExistingLabel' as ColumnKey<object, object>))
+        .toEqual('nonExistingLabel');
+    });
+  });
+
+  test('availableColumns should return every column without the "options." prefix', () => {
+    expect(component.availableColumns()).toEqual(['actions', 'duration', 'name']);
+  });
+
+  test('availableOptionsColumns should return every column with the "options." prefix', () => {
+    expect(component.availableOptionsColumns()).toEqual(['options.task_id']);
+  });
+
+  describe('updateColumn', () => {
+    it('should push a new column to the columnList if checked', () => {
+      component.updateColumn({checked: true} as MatCheckboxChange, 'duration' as ColumnKey<object, object>);
+      expect(component.columns).toEqual(['id', 'created_time', 'duration']);
+      expect(component.data.availableColumns).toEqual(['name', 'options.task_id', 'actions']);
+    });
+
+    it('should remove a column from the columnList if it is unchecked', () => {
+      component.updateColumn({checked: false} as MatCheckboxChange, 'id' as ColumnKey<object, object>);
+      expect(component.columns).toEqual(['created_time']);
+      expect(component.data.availableColumns).toEqual(['name', 'options.task_id', 'actions', 'id']);
+    });
+
+    it('should not push a column already in the column list', () => {
+      component.updateColumn({checked: true} as MatCheckboxChange, 'id' as ColumnKey<object, object>);
+      expect(component.columns).toEqual(['id', 'created_time']);
+    });
+
+    it('should not push a column already that is not in the available columns', () => {
+      component.updateColumn({checked: true} as MatCheckboxChange, 'someColumn' as ColumnKey<object, object>);
+      expect(component.columns).toEqual(['id', 'created_time']);
+    });
+
+    it('should not do anything if a column not in the column list is removed', () => {
```

Comment:
I didn't understand. 
Maybe you meant "should not do anything if a column is not in the column list" ?

Reply:
i will try to clarifiate it.